perf(index): remove `Idx` bounds-checks from `first` + `last` methods

overlookmotel opened the pull request:
The following methods of `IndexVec` call `Idx::from_usize` which performs a bounds check, with a panicking branch:

* `first`
* `first_mut`
* `last`
* `last_mut`

These bounds checks are unnecessary. Remove them by delegating to the underlying `Vec`'s methods.

overlookmotel commented:
* **#5726** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5726?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

IWANABETHATGUY commented on crates/oxc_index/src/idxslice.rs:
```diff
@@ -409,26 +409,25 @@ impl<I: Idx, T> IndexSlice<I, [T]> {
     #[inline]
     pub fn first_mut(&mut self) -> Option<&mut T> {
-        self.get_mut(I::from_usize(0))
+        self.raw.last_mut()
```
`self.raw.first_mut()`

overlookmotel replied:
Oh my god what an idiot I am! Thanks so much for spotting my error.

overlookmotel committed:
perf(index): remove `Idx` bounds-checks from `first` + `last` methods (#5726)

The following methods of `IndexVec` call `Idx::from_usize` which performs a bounds check, with a panicking branch:

* `first`
* `first_mut`
* `last`
* `last_mut`

These bounds checks are unnecessary. Remove them by delegating to the underlying `Vec`'s methods.
